src/main/webapp/scripts/session-script.js

```diff
@@ -1,20 +1,20 @@
 import { ServerClient } from './serverclient.js';
 
 /**
- * Represents the URLSearchParams of the
- * the client is in, holds information such as the
- * session ID and the screen name of the current user.
+ * Represents the URLSearchParams the client is in, 
+ * holds information such as the session ID and 
+ * the screen name of the current user.
  * @type {URLSearchParams}
  */
-const urlParameters = new URLSearchParams(window.location.search);
+let urlParameters;
 
 /**
  * Represents the ServerClient object responsible for
  * keeping up-to-date with the current session and handles many
- * of the client-to-server interactions, like passing the controller.
+ * of the client-to-server interactions, like changing the controller.
  * @type {ServerClient}
  */
-const client = new ServerClient(urlParameters);
+let client;
```

Comment:
lets call this serverClient, to be explicit about what it is a client to.

Client means different things to different people in different ways.